SIGSEGV when creating scrapers - "NaN is an unsupported value for field sum"

__Introduction:__
I try to setup multiple scrapers who each scrape metrics of different applications into their own buckets.
I didn't found an issue regarding SIGSEGV errors in scrapers so i'm opening this issue here.

__Steps to reproduce:__
1. Start a Docker Image of InfluxDB with the "latest" tag (InfluxDB 2.2.0)
2. Start a Docker Image for something to scrape e.g. promtail, loki, grafana, etc. with the "latest" tag
3. Wrap up the initial configuration of InfluxDB and create buckets for the metrics you want to scrape
4. Setup the applications you want to scrape
5. Create the scrapers for the metrics (e.g. http://grafana:3000/metrics, http://loki:3100/metrics, http://promtail:9080/metrics, etc.)
6. Watch everything go down in flames...
7. (Funfact: Scraping the metrics of InfluxDB itself via "http://influxdb:8086/metrics" works perfectly fine)

__Expected behavior:__
InfluxDB scraping the metrics of the applications into the created metrics buckets.

__Actual behavior:__
The container crashes and restarts (if configured) while throwing "segmentation violation" errors.

__Environment info:__
* ```Linux 5.17.11-300.fc36.x86_64 x86_64``` (Fedora 36 with latest packages as Host OS)
* ```InfluxDB v2.2.0 (git: a2f8538837) build_date: 2022-04-06T17:36:40Z``` (Output of "influxd version" inside the container)
* Docker runs as a rootless installation and the output of `docker version` looks as followed:
```
Client: Docker Engine - Community
 Version:           20.10.16
 API version:       1.41
 Go version:        go1.17.10
 Git commit:        aa7e414
 Built:             Thu May 12 09:16:46 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.16
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.10
  Git commit:       f756502
  Built:            Thu May 12 09:14:29 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.4
  GitCommit:        212e8b6fa2f44b9c21b2798135fc6fb7c53efc16
 runc:
  Version:          1.1.1
  GitCommit:        v1.1.1-0-g52de29d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

__Config:__
Some configurations were changed via environment variables in the docker-compose.yaml
```
- INFLUXD_QUERY_INITIAL_MEMORY_BYTES=10485760
- INFLUXD_QUERY_MAX_MEMORY_BYTES=1048576000
- INFLUXD_QUERY_MEMORY_BYTES=104857600
- INFLUXD_QUERY_CONCURRENCY=8
- INFLUXD_QUERY_QUEUE_SIZE=256
- INFLUXD_SESSION_LENGTH=60
- INFLUXD_LOG_LEVEL=info
```

__Logs:__
Depending on which scraper runs first, the logs look as followed:
```
influxdb  | ts=2022-05-29T18:11:05.922578Z lvl=info msg=Listening log_id=0alG4UBl000 service=tcp-listener transport=http addr=:8086 port=8086
influxdb  | ts=2022-05-29T18:11:05.922631Z lvl=info msg=Starting log_id=0alG4UBl000 service=telemetry interval=8h
influxdb  | ts=2022-05-29T18:11:15.925075Z lvl=error msg="Unable to gather list of points" log_id=0alG4UBl000 service=scraper scraper-name=monitoring-promtail error="NaN is an unsupported value for field sum"
influxdb  | panic: runtime error: invalid memory address or nil pointer dereference
influxdb  | [signal SIGSEGV: segmentation violation code=0x1 addr=0xb8 pc=0x19113db]
influxdb  | 
influxdb  | goroutine 391 [running]:
influxdb  | github.com/influxdata/influxdb/v2/v1/coordinator.(*PointsWriter).MapShards(0xc000f60180, 0xc00013d9e8)
influxdb  |     /root/project/v1/coordinator/points_writer.go:233 +0x69b
influxdb  | github.com/influxdata/influxdb/v2/v1/coordinator.(*PointsWriter).WritePointsPrivileged(0xc000f60180, {0x3dd24b0, 0xc000056050}, {0xc0014be000, 0x416347}, {0x21a7820, 0x7}, 0x1, {0xc0012e2000, 0xb6, ...})
influxdb  |     /root/project/v1/coordinator/points_writer.go:384 +0x1ae
influxdb  | github.com/influxdata/influxdb/v2/v1/coordinator.(*PointsWriter).WritePoints(0x0, {0x3dd24b0, 0xc000056050}, {0xc0014be000, 0x0}, {0x21a7820, 0xc002510480}, 0x1, {0x3de51b8, 0xc001178480}, ...)
influxdb  |     /root/project/v1/coordinator/points_writer.go:364 +0x5c
influxdb  | github.com/influxdata/influxdb/v2/storage.(*Engine).WritePoints(0xc000f5e000, {0x3dd24b0, 0xc000056050}, 0x0, 0x0, {0xc0012e2000, 0xb6, 0xb6})
influxdb  |     /root/project/storage/engine.go:278 +0x204
influxdb  | github.com/influxdata/influxdb/v2/replications.(*service).WritePoints(0xc0014cad80, {0x3dd24b0, 0xc000056050}, 0x79ff392a30dde2b3, 0x2ac03406c9e57d43, {0xc0012e2000, 0x1c, 0xb6})
influxdb  |     /root/project/replications/service.go:317 +0x2d0
influxdb  | github.com/influxdata/influxdb/v2/gather.(*Scheduler).doScrape.func1()
influxdb  |     /root/project/gather/scheduler.go:111 +0x5c8
influxdb  | created by github.com/influxdata/influxdb/v2/gather.(*Scheduler).doScrape
influxdb  |     /root/project/gather/scheduler.go:95 +0xcf
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | 2022-05-29T18:11:16.428552213Z      info    found existing boltdb file, skipping setup wrapper      {"system": "docker", "bolt_path": "/var/lib/influxdb2/influxd.bolt"}
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | 2022-05-29T18:11:16.542319275Z      info    found existing boltdb file, skipping setup wrapper      {"system": "docker", "bolt_path": "/var/lib/influxdb2/influxd.bolt"}
```
```
influxdb  | ts=2022-05-29T18:24:51.150489Z lvl=info msg=Listening log_id=0alGrqk0000 service=tcp-listener transport=http addr=:8086 port=8086
influxdb  | ts=2022-05-29T18:24:51.150544Z lvl=info msg=Starting log_id=0alGrqk0000 service=telemetry interval=8h
influxdb  | ts=2022-05-29T18:25:01.152165Z lvl=error msg="Unable to gather list of points" log_id=0alGrqk0000 service=scraper scraper-name=monitoring-loki error="NaN is an unsupported value for field sum"
influxdb  | ts=2022-05-29T18:25:01.152494Z lvl=error msg="Unable to gather list of points" log_id=0alGrqk0000 service=scraper scraper-name=monitoring-loki error="NaN is an unsupported value for field sum"
influxdb  | panic: runtime error: invalid memory address or nil pointer dereference
influxdb  | [signal SIGSEGV: segmentation violation code=0x1 addr=0xb8 pc=0x19113db]
influxdb  | 
influxdb  | goroutine 136 [running]:
influxdb  | github.com/influxdata/influxdb/v2/v1/coordinator.(*PointsWriter).MapShards(0xc0012421e0, 0xc00196f9e8)
influxdb  |     /root/project/v1/coordinator/points_writer.go:233 +0x69b
influxdb  | github.com/influxdata/influxdb/v2/v1/coordinator.(*PointsWriter).WritePointsPrivileged(0xc0012421e0, {0x3dd24b0, 0xc000056050}, {0xc00220bb60, 0x416347}, {0x21a7820, 0x7}, 0x1, {0xc000ad1500, 0x131, ...})
influxdb  |     /root/project/v1/coordinator/points_writer.go:384 +0x1ae
influxdb  | github.com/influxdata/influxdb/v2/v1/coordinator.(*PointsWriter).WritePoints(0x0, {0x3dd24b0, 0xc000056050}, {0xc00220bb60, 0x0}, {0x21a7820, 0xc0020239c0}, 0x1, {0x3de51b8, 0xc00221af60}, ...)
influxdb  |     /root/project/v1/coordinator/points_writer.go:364 +0x5c
influxdb  | github.com/influxdata/influxdb/v2/storage.(*Engine).WritePoints(0xc001232900, {0x3dd24b0, 0xc000056050}, 0x0, 0x0, {0xc000ad1500, 0x131, 0x131})
influxdb  |     /root/project/storage/engine.go:278 +0x204
influxdb  | github.com/influxdata/influxdb/v2/replications.(*service).WritePoints(0xc0018dcd80, {0x3dd24b0, 0xc000056050}, 0x79ff392a30dde2b3, 0x27665a2eb7ed2f88, {0xc000ad1500, 0x18, 0x131})
influxdb  |     /root/project/replications/service.go:317 +0x2d0
influxdb  | github.com/influxdata/influxdb/v2/gather.(*Scheduler).doScrape.func1()
influxdb  |     /root/project/gather/scheduler.go:111 +0x5c8
influxdb  | created by github.com/influxdata/influxdb/v2/gather.(*Scheduler).doScrape
influxdb  |     /root/project/gather/scheduler.go:95 +0xcf
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | 2022-05-29T18:25:01.663166111Z      info    found existing boltdb file, skipping setup wrapper      {"system": "docker", "bolt_path": "/var/lib/influxdb2/influxd.bolt"}
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
influxdb  | 2022-05-29T18:25:01.795336358Z      info    found existing boltdb file, skipping setup wrapper      {"system": "docker", "bolt_path": "/var/lib/influxdb2/influxd.bolt"}
```

Same results, trying to setup a scraper in Influxdb (via UI) and the instance crashes immediately.
Tried on v 2.4 and 2.5
```ts=2022-11-08T21:18:22.242168Z lvl=info msg="Opened file" log_id=0e2H_XG0000 service=storage-engine engine=tsm1 service=filestore path=/var/lib/influxdb2/engine/data/afc7cf371f4e9c81/autogen/289/000000004-000000002.tsm id=0 duration=0.396ms
ts=2022-11-08T21:18:22.242256Z lvl=info msg="Opened shard" log_id=0e2H_XG0000 service=storage-engine service=store op_name=tsdb_open index_version=tsi1 path=/var/lib/influxdb2/engine/data/afc7cf371f4e9c81/autogen/289 duration=7.640ms
ts=2022-11-08T21:18:22.248560Z lvl=info msg="Open store (end)" log_id=0e2H_XG0000 service=storage-engine service=store op_name=tsdb_open op_event=end op_elapsed=1243.344ms
ts=2022-11-08T21:18:22.248601Z lvl=info msg="Starting retention policy enforcement service" log_id=0e2H_XG0000 service=retention check_interval=30m
ts=2022-11-08T21:18:22.248623Z lvl=info msg="Starting precreation service" log_id=0e2H_XG0000 service=shard-precreation check_interval=10m advance_period=30m
ts=2022-11-08T21:18:22.249682Z lvl=info msg="Starting query controller" log_id=0e2H_XG0000 service=storage-reads concurrency_quota=1024 initial_memory_bytes_quota_per_query=9223372036854775807 memory_bytes_quota_per_query=9223372036854775807 max_memory_bytes=0 queue_size=1024
ts=2022-11-08T21:18:22.251845Z lvl=info msg="Configuring InfluxQL statement executor (zeros indicate unlimited)." log_id=0e2H_XG0000 max_select_point=0 max_select_series=0 max_select_buckets=0
ts=2022-11-08T21:18:22.262453Z lvl=info msg=Listening log_id=0e2H_XG0000 service=tcp-listener transport=http addr=:8086 port=8086
ts=2022-11-08T21:18:32.303884Z lvl=error msg="Unable to gather list of points" log_id=0e2H_XG0000 service=scraper scraper-name="Ads Grafana" error="NaN is an unsupported value for field sum"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb8 pc=0x7f816989539b]

goroutine 2486 [running]:
github.com/influxdata/influxdb/v2/v1/coordinator.(*PointsWriter).MapShards(0xc001a17980, 0xc0039839e0)
	/root/project/v1/coordinator/points_writer.go:233 +0x69b
github.com/influxdata/influxdb/v2/v1/coordinator.(*PointsWriter).WritePointsPrivileged(0xc001a17980, {0x7f816b480330?, 0xc000054030}, {0xc00c320be0, 0x10}, {0x7f816a1a23d7, 0x7}, 0x1?, {0xc0051e3a80, 0x1a3, ...})
	/root/project/v1/coordinator/points_writer.go:384 +0x1ab
github.com/influxdata/influxdb/v2/v1/coordinator.(*PointsWriter).WritePoints(0xc00c21cc00?, {0x7f816b480330?, 0xc000054030?}, {0xc00c320be0?, 0x0?}, {0x7f816a1a23d7?, 0x0?}, 0x9?, {0x7f816b48c7a8, 0xc00c317cb0}, ...)
	/root/project/v1/coordinator/points_writer.go:364 +0x5c
github.com/influxdata/influxdb/v2/storage.(*Engine).WritePoints(0xc0019f8900, {0x7f816b480330, 0xc000054030}, 0x1f?, 0x0?, {0xc0051e3a80, 0x1a3, 0x1a3})
	/root/project/storage/engine.go:278 +0x204
github.com/influxdata/influxdb/v2/replications.(*service).WritePoints(0xc012f2d560, {0x7f816b480330?, 0xc000054030}, 0xf9c8a855916a845d, 0x2b15100f84306656, {0xc0051e3a80?, 0x1a3, 0x1a3})
	/root/project/replications/service.go:323 +0xf7
github.com/influxdata/influxdb/v2/gather.(*Scheduler).doScrape.func1()
	/root/project/gather/scheduler.go:111 +0x5a4
created by github.com/influxdata/influxdb/v2/gather.(*Scheduler).doScrape
	/root/project/gather/scheduler.go:95 +0xca
```

--- a/gather/prometheus.go
+++ b/gather/prometheus.go
@@ -92,13 +92,25 @@ func (p *prometheusScraper) parse(r io.Reader, header http.Header, target influx
 			case dto.MetricType_SUMMARY:
 				// summary metric
 				fields = makeQuantiles(m)
-				fields["count"] = float64(m.GetSummary().GetSampleCount())
-				fields["sum"] = float64(m.GetSummary().GetSampleSum())
+				sc := float64(m.GetSummary().GetSampleCount())
+				ss := float64(m.GetSummary().GetSampleSum())
+				if !math.IsNaN(sc) {
+					fields["count"] = sc
+				}
+				if !math.IsNaN(ss) {
+					fields["sum"] = ss
+				}
 			case dto.MetricType_HISTOGRAM:
 				// histogram metric
 				fields = makeBuckets(m)
-				fields["count"] = float64(m.GetHistogram().GetSampleCount())
-				fields["sum"] = float64(m.GetHistogram().GetSampleSum())
+				sc := float64(m.GetHistogram().GetSampleCount())
+				ss := float64(m.GetHistogram().GetSampleSum())
+				if !math.IsNaN(sc) {
+					fields["count"] = sc
+				}
+				if !math.IsNaN(ss) {
+					fields["sum"] = ss
+				}
 			default:
 				// standard metric
 				fields = getNameAndValue(m)

--- a/gather/prometheus.go
+++ b/gather/prometheus.go
@@ -92,22 +92,18 @@ func (p *prometheusScraper) parse(r io.Reader, header http.Header, target influx
 			case dto.MetricType_SUMMARY:
 				// summary metric
 				fields = makeQuantiles(m)
-				sc := float64(m.GetSummary().GetSampleCount())
+				fields["count"] = float64(m.GetSummary().GetSampleCount())
+
 				ss := float64(m.GetSummary().GetSampleSum())
-				if !math.IsNaN(sc) {
-					fields["count"] = sc
-				}
 				if !math.IsNaN(ss) {
 					fields["sum"] = ss
 				}
 			case dto.MetricType_HISTOGRAM:
 				// histogram metric
 				fields = makeBuckets(m)
-				sc := float64(m.GetHistogram().GetSampleCount())
+				fields["count"] = float64(m.GetHistogram().GetSampleCount())
+
 				ss := float64(m.GetHistogram().GetSampleSum())
-				if !math.IsNaN(sc) {
-					fields["count"] = sc
-				}
 				if !math.IsNaN(ss) {
 					fields["sum"] = ss
 				}

